File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    participant_people = dict()
    completion_people = dict()
        
    for people in completion:
        completion_people.setdefault(people, 0)
        completion_people[people] += 1
    
    for people in participant:
        completion_people.setdefault(people, 0)
        if not completion_people[people]:
            return people
    
    for people in participant:
        participant_people.setdefault(people, 0)
        participant_people[people] += 1

    for people in participant:
        if participant_people[people] != completion_people[people]:
            return people

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    var myMap = new Map();
    
    for (const part of participant) {
        if (!myMap.get(part)) {
            myMap.set(part, 1)
        } else {
            myMap.set(part, myMap.get(part) + 1)
        }
    }
    
    for (const comple of completion) {
        myMap.set(comple, myMap.get(comple) - 1)
    }
    
    var answer = '';
    
    for (const part of participant) {
        if (myMap.get(part)) {
            answer = part;
            break
        }
    }
    
    
    return answer;
}